build_wheel.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+#************************************************#
+#           build_wheel.sh                       #
+#    Build a Python3 wheel for YOLACT            #
+#************************************************#
+
+REPO_NAME="yolact"
+NEEDED_FOLDERS=("data" "external" "layers" "scripts" "utils" "web")
+REQUIREMENTS_FILE="requirements.txt"
+SETUP_FILE="setup.py"
+PYTHON_EXTENSION=".py"
+PACKAGE_TEST_DIR="/tmp/$USER/pip_package_test"
+
+# --------------------------------------------------------- #
+# main ()                                                   #
+# drives the entire build process                           #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+main() {
+  echo "Starting build ..."
+  pre_build_setup &&
+    build &&
+    post_build_test_and_teardown &&
+    echo "Build Successful ... "
+}
+
+# --------------------------------------------------------- #
+# pre_build_setup ()                                        #
+# performs necessary setup for building a YOLACT wheel      #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+pre_build_setup() {
+  echo "Pre-Build Setup ... "
+  __validate_files && __copy_files && __fix_requirements && __build_setup
+}
+
+# --------------------------------------------------------- #
+# build ()                                                  #
+# actual build happens here                                 #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+build() {
+  echo "Build ... "
+  python3 -m build
+  rm --recursive "${REPO_NAME}"
+}
+
+# --------------------------------------------------------- #
+# post_build_test_and_teardown ()                           #
+# installation check + cleanup unnecessary files            #
+# Returns: 0 on success, non-zero if something goes wrong.  #
+# --------------------------------------------------------- #
+post_build_test_and_teardown() {
+  echo "Post-Build Test and Teardowns ... "
+  __test_build
+  __teardown
+}
+
+__validate_files() {
+  echo "Validating files ... "
+  # venv
+  if [[ ! -e "$(command -v python3)" ]]; then
+    echo "Is python installed?" 1>&2
+    return 1
+  fi
+
+  # setup.py
+  if [[ ! -e "${SETUP_FILE}" ]]; then
+    echo "${SETUP_FILE} file not found" 1>&2
+    return 3
+  fi
+}
+
+__copy_files() {
+  echo "Copying files ... "
+
+  # create empty yolact dir, overwrite if exists
+  [[ -d ${REPO_NAME} ]] && rm --recursive ${REPO_NAME}
```

Comment:
YMMV, but I think this would be equivalent and maybe more human friendly?
```bash
test -d ${REPO_NAME} && rm --recursive ${REPO_NAME}
```

Also, you could consider just `rm -rf ${REPO_NAME}` since `rm` will exit with a 0 status code if you try to forcefully delete a non-existent directory:
```bash
$ rm -rf a_folder
$ echo $?
0
```

Reply:
It's not important to me, but you don't quote `${REPO_NAME}` on this line, while you do quote it elsewhere.